sdk/storage/storage-file-datalake/test/perfstress/track-2/uploadFromFile.spec.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { StorageDFSUploadTest } from "./upload.spec";
+import fs from "fs";
+import util from "util";
+const writeFile = util.promisify(fs.writeFile);
+const fileExists = util.promisify(fs.exists);
+const mkdir = util.promisify(fs.mkdir);
+const deleteFile = util.promisify(fs.unlink);
+
+const localDirName = "temp";
+const localFileName = `${localDirName}/upload-from-test-temp-file.txt`;
+
+export class StorageDFSUploadFromFileTest extends StorageDFSUploadTest {
+  public async globalSetup() {
+    await super.globalSetup();
+    if (!(await fileExists(localDirName))) await mkdir(localDirName);
+    await writeFile(localFileName, Buffer.alloc(this.parsedOptions.size.value!));
+  }
+
+  public async globalCleanup() {
+    await super.globalCleanup();
+    await deleteFile(localFileName);
+  }
+
+  async runAsync(): Promise<void> {
+    // ${Math.floor(Math.random() * 1000)}
+    // so that the following service error is not seen
+    // Error"The uploaded data is not contiguous or the position query parameter value is not equal to the length of the file after appending the uploaded data."
+    await this.directoryClient
+      .getFileClient(`newfile${new Date().getTime()}-${Math.floor(Math.random() * 1000)}`)
```

Comment:
Similar to the "upload" test, I think each instance should be able to upload to the same file name repeatedly.  I would only expect an error if two instances try to upload to the same file name in parallel.  I would also recommend using GUIDs instead of timestamps or random numbers if we need something truly unique.